From 1d01cd12a4f4dce61ed47702da63e1ee5cb20fef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juho=20H=C3=A4m=C3=A4l=C3=A4inen?=
 <juho.hamalainen@tieto.com>
Date: Wed, 4 Mar 2015 12:09:49 +0200
Subject: [PATCH 10/15] Do not try to acquire resources in setMedia() in
 mediaplayer.

Acquire() is called in play() anyway.

Change-Id: I4c6688006d74490d774813b3f1fb381d915ffe23
---
 src/plugins/gstreamer/mediaplayer/qgstreamerplayercontrol.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/plugins/gstreamer/mediaplayer/qgstreamerplayercontrol.cpp b/src/plugins/gstreamer/mediaplayer/qgstreamerplayercontrol.cpp
index ffb36e7e..4098f79a 100644
--- a/src/plugins/gstreamer/mediaplayer/qgstreamerplayercontrol.cpp
+++ b/src/plugins/gstreamer/mediaplayer/qgstreamerplayercontrol.cpp
@@ -371,12 +371,8 @@ void QGstreamerPlayerControl::setMedia(const QMediaContent &content, QIODevice *
     m_session->showPrerollFrames(false); // do not show prerolled frames until pause() or play() explicitly called
     m_setMediaPending = false;
 
-    if (!content.isNull() || stream) {
-        if (!m_resources->isGranted())
-            m_resources->acquire();
-    } else {
+    if (content.isNull() && !stream)
         m_resources->release();
-    }
 
     m_session->stop();
 
-- 
2.15.0

